Document new flexible demand and supply behaviour

antw opened the pull request:
This updates the documentation for flexibility to reflect the changes that were made during the price-sensitive flex project. I'm confident that it covers the changes we made in quite a bit of detail; thanks @mabijkerk! If you're also happy, then I think we can merge this.

Closes #101.

mabijkerk commented:
I have made some minor changes, including updating the description of price-sensitive demands in the Merit Order page.
There are still two questions I have about that page, which I was hoping you could have look at @antw:

1) Part of the description refers to a Figure which no longer seems to be present in the documentation.
> The module starts with the cheapest plant first (lowest marginal operation costs). In Figure 6., this would be the yellow arrow (1). 

2) The description of price-setting by the Merit Order Module seems to suggest that price-setting only be done by the marginal power plant. In the price-sensitive-flex project it also became possible for flexible demand technologies to set the price right?
> Thirdly, the Merit Order Module sets an electricity price for each hour. The plant with the highest operating cost that will come online last sets the price for electricity.

antw commented:
> Part of the description refers to a Figure which no longer seems to be present in the documentation.

It was still on the page, just quite far away from the text and no longer labelled as "figure 6". I've moved it closer and linked to it from the text.

> In the price-sensitive-flex project it also became possible for flexible demand technologies to set the price right?

Good catch. I've updated this.

mabijkerk reviewed (approved):
Nice work!
